### PR TITLE
More testing, esp core layers

### PR DESCRIPTION
**Pull request by phreeza**
Making progress on the tests slowly but surely. Here is another intermediate PR, brings test coverage to ~60%.

### CODE REVIEW

**Comment by phreeza** [comment]
This should ideally be merged together with #407

File: tests/auto/keras/layers/test_core.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    Test the constructor, config and params functions of all layers in core.
+    """
+
+    def _runner(self, layer):
+        conf = layer.get_config()
+        assert (type(conf) == dict)
+
+        param = layer.get_params()
+        # Typically a list or a tuple, but may be any iterable
+        assert hasattr(param, '__iter__')
```
Comment (fchollet) [inline tests/auto/keras/layers/test_core.py]: `get_params` is supposed to return a tuple of lists `theano_tensors, regularizers, constraints`, with `theano_tensors` and `constraints` having the same length. We should definitely check the types and the length condition.

Comment (fchollet) [inline tests/auto/keras/layers/test_core.py]: We should also test `set_weights`, `get_weights`, `get_ouput`, `get_input`, `set_previous`, `supports_masked_input` in this method, via Theano compilation of the layer in an expression...

Comment (fchollet) [inline tests/auto/keras/layers/test_core.py]: (never mind my previous comment, it seems testing of other layer methods is handled elsewhere)

**Comment by fchollet** [comment]
Cool, looking good!

File: tests/auto/keras/layers/test_core.py (hunk 1 above)
Comment (phreeza) [inline tests/auto/keras/layers/test_core.py]: I will make sure to check that assumption in a test, added to my todo list.

**Pull request merged**